packages/web/modals/profile.tsx

```diff
@@ -174,10 +348,213 @@ export const ProfileModal: FunctionComponent<
+                                              : "text-osmoverse-300"
+                                          }`
+                                        }
+                                        key={collection.collection.name}
+                                        value={collection.collection.name}
+                                      >
+                                        <div className="text-center">
+                                          <p className="subtitle1 tracking-wide text-osmoverse-300">
+                                            {collection.collection.name}
+                                          </p>
+                                        </div>
+                                      </Listbox.Option>
+                                    )
+                                  )}
+                                </Listbox.Options>
+                              </div>
+                            </Listbox>
+                            <div className="relative mt-2 flex flex-col pr-4">
+                              {filteredTokens?.filter(
+                                (nft) => nft?.media?.type === "image"
+                              ).length > 0 && (
+                                <div className="max-w-64 relative mt-4 w-full cursor-default rounded-lg bg-osmoverse-600 py-2 pl-4 pr-10 text-left shadow-md focus:outline-none focus-visible:border-osmoverse-300 focus-visible:ring-2 focus-visible:ring-white-full focus-visible:ring-opacity-75 focus-visible:ring-offset-2 focus-visible:ring-offset-wosmongton-300 sm:w-full sm:text-sm">
+                                  <input
+                                    type="text"
+                                    value={searchTokenValue}
+                                    onChange={(e) =>
+                                      setSearchTokenValue(e.target.value)
+                                    }
+                                    onClick={(e) => {
+                                      e.preventDefault();
+                                      e.stopPropagation();
+                                    }}
+                                    className="w-full bg-transparent placeholder-osmoverse-200 outline-none"
+                                    placeholder={"Type to search for a token"}
+                                  />
+                                  <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+                                    <Icon
+                                      className="flex shrink-0 items-center text-white-full"
+                                      id={"search"}
+                                      height={12}
+                                      width={12}
+                                    />
+                                  </span>
+                                </div>
+                              )}
+                              {filteredTokens?.length > 0 &&
```

Comment:
Could pull this into it's own variables since it's been used 3 times

const tokensWithImage = filteredTokens?.filter((nft) => nft?.media?.type === "image")